src/client.ts

```diff
@@ -0,0 +1,71 @@
+import { VannaFeature } from "./types";
+import { VannaSource } from "./source";
+
+interface Mapping<T> {
+  [key: string]: T;
+}
+
+type VannaFeatureMapping = Mapping<VannaFeature>;
+
+export function mergeFeatures(
+  featuresList: VannaFeature[][]
+): VannaFeatureMapping {
+  const mapping: VannaFeatureMapping = {};
+  featuresList.forEach(features => {
+    features.forEach(feature => {
+      mapping[feature.slug] = feature;
+    });
+  });
+
+  return mapping;
+}
+
+export function resolveVariation(feature: VannaFeature, options: VannaOptions) {
+  const invalidTarget = feature.targets && !options.target;
+  if (invalidTarget) {
+    throw new Error(
+      "Feature has targets, but no target was passed when FeatureClient was setup"
+    );
+  }
+
+  if (feature.type === "boolean") {
+    if (feature.targets) {
+      return feature.targets.includes(options.target || "");
```

Comment:
Do you need the `|| ""` guard when you have the guard on line 25? Could this line read:

`return feature.targets.includes(options.target);`

Reply:
Yeah, it's purely to pass the type. It shouldn't reach based on the check on `feature.targets && !options.target`